### ISSUE TITLE
Docker images based on image 3scale/base:precise fail to use Rubygems

### ISSUE THREAD

**Issue by miguelsorianod** [3scale/docker-base#2]
It seems that Rubygems is no longer accessible when using the current 3scale/base:precise image from quay.io:

```
Fetching source index from https://rubygems.org/
Retrying source fetch due to error (2/3): Bundler::HTTPError Could not fetch specs from https://rubygems.org/
Retrying source fetch due to error (3/3): Bundler::HTTPError Could not fetch specs from https://rubygems.org/
Could not fetch specs from https://rubygems.org/
```

This is because Rubygems deprecated TLSv1.0 and TLSv1.1 protocols some time ago: https://blog.rubygems.org/2018/02/24/tls-10-and-11-deprecation-notice.html
This means that TLSv1.2 is needed to access Rubygems.

precise is Ubuntu 12.04. In Ubuntu the available openssl versions is 1.0.1, which includes support for TLSv1.2. However, Ubuntu disabled TLSv1.2 from openssl in a patchset in its day due to was not correctly handled by most servers: https://bugs.launchpad.net/ubuntu/+source/openssl/+bug/1256576 (see #4).
After some time they reenabled TLSv1.2 again in a newer patchset: https://usn.ubuntu.com/2606-1/

I guess to fix this a new version of the 3scale/base:precise public image should be generated hoping that ubuntu:precise contains the openssl libraries updates.

I open this case because I understand the 3scale/base:precise is a dependency of other images so the consequences of doing this should be evaluated

cc @mikz 

Also cc @maneta because this does not happen only in entities that use the Docker image, but in ALL machines that have Ubuntu 12.04 and have installed openssl/libssl versions older than 1.0.1-4ubuntu5.27 If you want I can open an internal issue with details on how to check if a machine has this problem and other information.

**Comment by mikz** [comment]
@miguelsorianod so latest openssl libraries in precise should be fine? We can just rebuild the image ?

**Comment by maneta** [comment]
@miguelsorianod AFAIK @orimarti already have corrected this in some other images.

**Comment by miguelsorianod** [comment]
Hi @mikz,

Yes, the latest openssl libraries in precise should be fine. By just rebuilding the images we would have the latest libssl available. I verified that ubuntu:precise comes with libssl 1.0.1-4ubuntu5.39, that has TLSv1.2 enabled:

ii  libssl1.0.0                 1.0.1-4ubuntu5.39            SSL shared libraries

My main concern was all the other images that depend on the 3scale/base:precise, that would also have to be recreated again, because this image has not been updated in a long time.

**Comment by mikz** [comment]
@miguelsorianod we can rebuild only images we have issues with.

**Comment by miguelsorianod** [comment]
I have generated and uploaded to quay the new docker image for precise that contains the most up-to-date libssl libraries available in Ubuntu 12.04. It is publicly available in: quay.io/3scale/base:precise with hash:
sha256:c0c8ca2a76416f7110538895aaf3fcce1d2155d108ed151306c1495544bb5fb8

When the other images based on this one are regenerated they should incorporate the changes.

I close this issue.
